Xamarin.PropertyEditing.Mac/Controls/CollectionInlineEditorControl.cs

```diff
@@ -25,16 +28,30 @@ public CollectionInlineEditorControl (IHostResourceProvider hostResources)
 			};
 
 			this.openCollection.Activated += (o, e) => {
+
+				openCollection.Enabled = false;
+				var parentWindow = Window;
+				
 				var w = new CollectionEditorWindow (hostResources, ViewModel) {
 					Appearance = EffectiveAppearance
 				};
 
-				CocoaHelpers.RunModalForWindow (w, this.openCollection, responseHandler: result => {
-					if (result != NSModalResponse.OK)
-						ViewModel.CancelCommand.Execute (null);
-					else
-						ViewModel.CommitCommand.Execute (null);
-				});
+				var result = (NSModalResponse)(int)NSApplication.SharedApplication.RunModalForWindow (w);
+
+				//after run modal our FocusedWindow is null, we set the parent again
+				parentWindow?.MakeKeyAndOrderFront (parentWindow);
+
+				if (result != NSModalResponse.OK)
+					ViewModel.CancelCommand.Execute (null);
+				else
+					ViewModel.CommitCommand.Execute (null);
+
+				//small hack to override vs4mac default focus
+				System.Threading.Tasks.Task.Delay (DefaultDelayTime)
+				.ContinueWith (t => {
+					openCollection.Enabled = true;
```

Comment:
As per previous comment, we shouldn't need this, based on VS4M testing.

Reply:
fixed